packages/dds/tree/src/test/simple-tree/getJsonSchema.spec.ts

```diff
@@ -15,6 +15,37 @@ import { hydrate } from "./utils.js";
 import { getJsonValidator } from "./jsonSchemaUtilities.js";
 
 describe("getJsonSchema", () => {
+	it("Field Schema", async () => {
+		const schemaFactory = new SchemaFactory("test");
+		const Schema = schemaFactory.optional(schemaFactory.string, {
+			metadata: { description: "An optional string." },
+		});
+
+		const actual = getJsonSchema(Schema);
+
+		const expected: JsonTreeSchema = {
+			$defs: {
+				"com.fluidframework.leaf.string": {
+					type: "string",
+					_treeNodeSchemaKind: NodeKind.Leaf,
+				},
+			},
+			$ref: "#/$defs/com.fluidframework.leaf.string",
+		};
```

Comment:
Tangent: this doesn't seem to indicate that the string is optional, correct? If it does, I don't understand how, so I was wondering for my own education. But also, on line 46 it seems like we're validating that assigning `null` isn't valid. Should we validate what happens when assigning `undefined`?

Reply:
Required fields are included at the end of a definition like so:

```json
{
  "$id": "https://example.com/address.schema.json",
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "description": "An address similar to http://microformats.org/wiki/h-card",
  "type": "object",
  "properties": {
    "postOfficeBox": {
      "type": "string"
    },
    "extendedAddress": {
      "type": "string"
    },
    "streetAddress": {
      "type": "string"
    },
    "locality": {
      "type": "string"
    },
    "region": {
      "type": "string"
    },
    "postalCode": {
      "type": "string"
    },
    "countryName": {
      "type": "string"
    }
  },
  "required": [ "locality", "region", "countryName" ],
  "dependentRequired": {
    "postOfficeBox": [ "streetAddress" ],
    "extendedAddress": [ "streetAddress" ]
  }
}
```
So, I would imagine some kind of a test here would check for the existence of the 'required' field and its values right?

Reply:
Our unit tests compare exact JSON output, so this is covered. But yes, JSON Schema validators check the `required` property for this. Items not enumerated in that property are implicitly optional.